From 9251789778ad3ced2ed269fdf487e11579b05ba9 Mon Sep 17 00:00:00 2001
From: Keerthana K <keerthanak@vmware.com>
Date: Thu, 11 Nov 2021 05:59:05 +0000
Subject: [PATCH 6/9] stalld: Fix single threaded mode starvation threshold
 check

starvation check for tasks in single threaded mode should be
>= config_starving_threshold. Otherwise, a task which is
starved for config_starving_threshold secs will not get boosted
unless it starves for atleast config_starving_threshold + config_granularity
secs if there was no boost happened or config_starving_threshold +
(config_granularity - config_boost_duration) if there was a boost happened.

[sshedi: port to stalld-1.17.1]

Signed-off-by: Keerthana K <keerthanak@vmware.com>
Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 src/stalld.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/stalld.c b/src/stalld.c
index 1716e5b..f040815 100644
--- a/src/stalld.c
+++ b/src/stalld.c
@@ -1146,7 +1146,7 @@ int boost_cpu_starving_vector(struct cpu_starving_task_info *vector, int nr_cpus
 		if (config_log_only)
 			continue;
 
-		if (cpu->pid != 0 && (now - cpu->since) > config_starving_threshold) {
+		if (cpu->pid != 0 && (now - cpu->since) >= config_starving_threshold) {
 			/*
 			 * Check if this task name is part of a denylist
 			 * If yes, do not boost it.
-- 
2.39.0

